azurerm/internal/services/web/resource_arm_app_service_certificate.go

```diff
@@ -130,8 +136,16 @@ func resourceArmAppServiceCertificateCreateUpdate(d *schema.ResourceData, meta i
 	pfxBlob := d.Get("pfx_blob").(string)
 	password := d.Get("password").(string)
 	keyVaultSecretId := d.Get("key_vault_secret_id").(string)
+	hostingEnvironmentProfileId := d.Get("hosting_environment_profile_id").(string)
 	t := d.Get("tags").(map[string]interface{})
 
+	var hep *web.HostingEnvironmentProfile
+	if len(hostingEnvironmentProfileId) > 0 {
+		hep = &web.HostingEnvironmentProfile{
+			ID: &hostingEnvironmentProfileId,
+		}
+	}
```

Comment:
if we move this below the creation of the certificate var
```suggestion
	if len(hostingEnvironmentProfileId) > 0 {
		certificate.CertificateProperties = &web.HostingEnvironmentProfile{
			ID: &hostingEnvironmentProfileId,
		}
	}
```

Reply:
I moved the creation, as suggested, and updated the docs, but it'll take me a bit to dig in to your testing to find good tests for this change.

Reply:
On second thought, you sure you want some sort of positive test?  I'm pretty sure that to do so would require spinning up an App Service Environment and those aren't cheap or quick (or easy to tear down).  Example:

Normal Web App Cert:
```
{
...
  "properties": {
  ...
    "hostingEnvironmentProfile": null,
  }
}
```

ASE Web App Cert:
```
{
...
  "properties": {
  ...
    "hostingEnvironmentProfile": {
      "id": "/subscriptions/<subId>/resourceGroups/<rgName>/providers/Microsoft.Web/hostingEnvironments/<ASEName>",
      "name": "<ASEName>",
      "type": "Microsoft.Web/hostingEnvironments"
    },
  }
}
```
If so, I'm game to try, but new to your testing frameworks so any pointers you could provide would be appreciated.

Reply:
@wmdave - you can just take an existing ASE and cert test and then combine them. cost isn't really an issue for us as much as ensuring the functionality works, and doesn't break in the future

Reply:
Okay, I think it's ready to go.  Changed the format of the block as described in a subsequent comment, added test, updated site docs and did terrafmt.  Let me know if anything doesn't look right.  Sorry about the delay in getting it all going.

Reply:
Oh yeah, sorry, dhdanie = wmdave.  Didn't realize I was logged in to a different account.